Frameworks/Foundation/NSDecimalNumber.mm

```diff
@@ -14,296 +14,328 @@
+    });
+    return _maxNumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalMultiplyByPowerOf10(NSDecimal* result, const NSDecimal* number, short power, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)minimumDecimalNumber {
+    static dispatch_once_t __minNumberOnce;
+    dispatch_once(&__minNumberOnce, ^{
+        _minNumber = [[NSDecimalNumber alloc] initWithMantissa:std::numeric_limits<unsigned long long>::max() exponent:127 isNegative:YES];
+    });
+    return _minNumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalNormalize(NSDecimal* number1, NSDecimal* number2, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)decimalNumberWithDecimal:(NSDecimal)decimal {
+    return [[[NSDecimalNumber alloc] initWithDecimal:decimal] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
++ (NSDecimalNumber*)decimalNumberWithMantissa:(unsigned long long)mantissa exponent:(short)exponent isNegative:(BOOL)isNegative {
+    return [[[NSDecimalNumber alloc] initWithMantissa:mantissa exponent:exponent isNegative:isNegative] autorelease];
 }
 
 /**
  @Status Stub
  @Notes
 */
-NSCalculationError NSDecimalPower(NSDecimal* result, const NSDecimal* number, NSUInteger power, NSRoundingMode roundingMode) {
++ (NSDecimalNumber*)decimalNumberWithString:(NSString*)numericString locale:(id)locale {
     UNIMPLEMENTED();
     return StubReturn();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-void NSDecimalRound(NSDecimal* result, const NSDecimal* number, NSInteger scale, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
+- (NSNumber*)initWithBool:(BOOL)value {
+    return [self initWithLongLong:static_cast<long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSString* NSDecimalString(const NSDecimal* dcm, id locale) {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithChar:(char)value {
+    return [self initWithLongLong:static_cast<long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalSubtract(NSDecimal* result,
-                                     const NSDecimal* leftOperand,
-                                     const NSDecimal* rightOperand,
-                                     NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithDouble:(double)value {
+    int exp = 0;
+    double res = value;
+    if (res < 0) {
+        res *= -1;
+    }
+
+    if (res > 1.8e146) {
+        return [self initWithMantissa:ULONG_LONG_MAX exponent:0x7f isNegative:(value < 0)];
+    }
+    // Attempt best for smaller numbers (this will be compacted later on)
+    while ((res != 0) && (res < static_cast<double>(ULLONG_MAX))) {
```

Comment:
Code comments without tasks? Are you planning on fixing this up in this review or addressing this later?